view/frontend/templates/hyva/bfcache/handler.phtml

```diff
@@ -10,164 +10,169 @@ use MageOS\ThemeOptimization\ViewModel\BfCache;
-                    cartDrawerData.open = false;
-                    document.body.style.overflow = '';
-                }
-
-                if (autoCloseMenuMobile) {
-                    this.handleMenuClosure();
+        <?php
+        /**
+         * Handle closing of mobile & desktop menus via Alpine.js
+         */ ?>
+        handleMenuClosure() {
+            if (typeof Alpine === 'undefined' || !Alpine.$data) return;
+
+            const mobileMenu = this.getElement('mobileMenu');
+            if (mobileMenu) {
+                const mobileMenuData = Alpine.$data(mobileMenu);
+                if (mobileMenuData?.open !== undefined) {
+                    mobileMenuData.open = false;
                 }
             }
 
-            /**
-             * Close mobile and desktop menus
-             * Uses Alpine.js data binding to manage menu states
-             */
-            handleMenuClosure() {
-                const mobileMenu = document.querySelector("[x-data^='initMenuMobile']");
-                if (mobileMenu && typeof Alpine !== 'undefined') {
-                    const mobileMenuData = Alpine.\$data(mobileMenu);
-                    if (mobileMenuData && typeof mobileMenuData.open !== 'undefined') {
-                        mobileMenuData.open = false;
-                    }
-                }
-
-                const desktopMenu = document.querySelector("[x-data^='initMenuDesktop']");
-                if (desktopMenu && typeof Alpine !== 'undefined') {
-                    const desktopMenuData = Alpine.\$data(desktopMenu);
-                    if (desktopMenuData && typeof desktopMenuData.hoverPanelActiveId !== 'undefined') {
-                        desktopMenuData.hoverPanelActiveId = 0;
-                    }
+            const desktopMenu = this.getElement('desktopMenu');
+            if (desktopMenu) {
+                const desktopMenuData = Alpine.$data(desktopMenu);
+                if (desktopMenuData?.hoverPanelActiveId !== undefined) {
+                    desktopMenuData.hoverPanelActiveId = 0;
                 }
             }
-        };
-
-        /**
-         * Handle pageshow event for BFCache scenarios
-         * Detects back/forward navigation and page restoration from BFCache
-         */
-        const handlePageShow = (event) => {
-            if (event.persisted) {
-                new BFCacheHandler().init();
-            }
-        };
-
-        // Register event listener with proper cleanup capability
-        window.addEventListener('pageshow', handlePageShow, { passive: true });
-    })();
+        }
+    }
+
+    <?php
+    /**
+     * Page restoration handler — persistent global instance
+     */ ?>
+    window.addEventListener('pageshow', (event) => {
+        if (event.persisted) {
+            (window._bfcacheHandler ??= new BFCacheHandler()).init();
+        }
+    }, { passive: true });
+})();
 </script>
+
 <?php $hyvaCsp->registerInlineScript() ?>
```

Comment:
@5mehulhelp5
For me everything good.
Only small issue: can you add this line right after close script tag, refer to this document https://docs.hyva.io/hyva-themes/writing-code/csp/csp-compatibility.html#how-does-hyva-work-without-unsafe-inline
```
The $hyvaCsp->registerInlineScript() has to be called directly after the closing script tag!
There must be no other HTML in between.
```

Reply:
I've added that warning